bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java

```diff
@@ -157,6 +157,8 @@ public void removeBindingConfigReader(BindingConfigReader reader) {
                 }
             }
         }
+        itemsMap.put(modelName, items);
```

Comment:
This looks like a brute force fix to me. Note that the itemsMap is also filled in `modelChanged`, so you are now doing it several times.
Wouldn't the better place the `getAll()` method as this is the one that gets called (once) before added events are received? And if this would additionally check, whether an entry in the map already exists, it would be even nicer.

Reply:
You are right and in addition the line introduces just another bug when multiple *.items models are present. Only one model would be written to itemsMap. Update is on the way.

Reply:
Oh noes, wrote that before I had the first coffee... Its a map idiot (<- note to myself).